src/hast2jcr/handlers/block.js

```diff
@@ -79,18 +79,107 @@ function collapseField(id, fields, properties, node) {
+      }
+      return name;
+    }, item.name);
+
+    return !(itemNames.includes(itemNameWithoutSuffix) && itemNameWithoutSuffix !== item.name);
+  });
+}
+
+function createComponentGroups(fields) {
+  const components = [];
+  fields.forEach((obj) => {
+    if (obj.name.includes('_')) {
+      const groupName = obj.name.split('_')[0];
+      let groupObj = components.find((item) => item.name === groupName);
+      if (!groupObj) {
+        groupObj = {
+          component: 'group',
+          name: groupName,
+          fields: [],
+        };
+        components.push(groupObj);
+      }
+      groupObj.fields.push(obj);
+    } else {
+      components.push(obj);
+    }
+  });
+  return components;
+}
+
+function isHeadline(handler, field, fields) {
+  return handler.name === 'title' && field.component === 'text' && fields.find((f) => f.name === `${field.name}Type`);
+}
+
+function findFieldByType(handler, groupFields, fields, idx) {
+  let groupField = null;
+  let gIdx = idx;
+  for (let index = gIdx; index < groupFields.length; index += 1) {
+    const field = groupFields[index];
+    if (field.component === handler.name
+      || isHeadline(handler, field, fields)
+      || (field.component === 'richtext' && handler.name === 'text')
+      || (field.component === 'reference' && handler.name === 'button')) {
+      groupField = field;
+      if (field.component === 'richtext' && handler.name === 'text') {
+        gIdx = index;
+      } else {
+        gIdx = index + 1;
+      }
+
+      break;
+    }
+  }
+  return { gIdx, groupField };
+}
+
+function extractProperties(node, id, ctx, mode = 'container') {
   const children = node.children.filter((child) => child.type === 'element');
   const properties = {};
-  const fields = findFieldsById(componentModels, id);
+  const { componentModels } = ctx;
+  const fields = createComponentGroups(findFieldsById(componentModels, id));
   if (!fields) {
     return properties;
   }
-  fields.forEach((field, idx) => {
+  const mainFields = getMainFields(fields);
+  mainFields.forEach((field, idx) => {
     if (children.length <= idx) {
       return;
     }
-    if (field.name === 'classes') {
+    if (field.component === 'group') {
+      const groupFields = getMainFields(field.fields);
+      const containerNode = select('div > div', children[idx]);
+      const containerChildren = containerNode.children.filter((child) => child.type === 'element');
+      let groupFieldIdx = 0;
+      containerChildren.forEach((containerChild) => {
+        const handler = getHandler(containerChild, [node], ctx);
+        if (handler) {
+          const {
+            gIdx,
+            groupField,
+          } = findFieldByType(handler, groupFields, field.fields, groupFieldIdx);
+          if (groupField) {
+            if (properties[groupField.name]) {
+              properties[groupField.name] = encodeHtml(`<p>${properties[groupField.name]}<p><p>${toString(containerChild).trim()}</p>`);
```

Comment:
iiuc this concatenates the previous value of `${properties[groupField.name]}` with the next value. 

that works fine for 2, but what happens with 2 paragraphs. It will wrap the previous `<p>...</p><p>...</p>` in another `<p>` if I'm not mistaken.